metron-platform/metron-management/src/main/java/org/apache/metron/management/FileSystemFunctions.java

```diff
@@ -0,0 +1,402 @@
+        String message = "Unable to read " + path + ": " + e.getMessage();
+        LOG.error(message, e);
+        return null;
+      }
+    }
+  }
+
+  static class FileSystemGet extends FileSystemFunction {
+
+    FileSystemGet(FileSystemGetter getter) {
+      super(getter);
+    }
+
+    @Override
+    public Object apply(List<Object> args, Context context) throws ParseException {
+      String path = (String) args.get(0);
+      if(path == null) {
+        return null;
+      }
+      try(FSDataInputStream is = fs.open(new Path(path))) {
+        return IOUtils.toString(is);
+      } catch (IOException e) {
+        String message = "Unable to read " + path + ": " + e.getMessage();
+        LOG.error(message, e);
+        return null;
+      }
+    }
+  }
+
+  static class FileSystemRm extends FileSystemFunction {
+
+    FileSystemRm(FileSystemGetter getter) {
+      super(getter);
+    }
+
+    @Override
+    public Object apply(List<Object> args, Context context) throws ParseException {
+
+      String path = (String) args.get(0);
+      if(path == null) {
+        return false;
+      }
+
+      boolean recursive = false;
+      if(args.size() > 1) {
+        recursive = ConversionUtils.convert(args.get(1), Boolean.class);
+      }
+
+      try {
+        fs.delete(new Path(path), recursive);
+        return true;
+      } catch (IOException e) {
+        String message = "Unable to remove " + path + (recursive?" recursively":"") + ": " + e.getMessage();
+        LOG.error(message, e);
+        return false;
+      }
+    }
+  }
+
+  static class FileSystemPut extends FileSystemFunction {
+
+    FileSystemPut(FileSystemGetter getter) {
+      super(getter);
+    }
+
+    @Override
+    public Object apply(List<Object> args, Context context) throws ParseException {
+      String content = (String)args.get(0);
+      if(content == null) {
+        return false;
+      }
+      String path = (String) args.get(1);
+      if(path == null) {
+        return false;
+      }
+
+      try(FSDataOutputStream os = fs.create(new Path(path))) {
+        os.writeBytes(content);
+        os.flush();
+        return true;
+      } catch (IOException e) {
+        String message = "Unable to write " + path + ": " + e.getMessage();
+        LOG.error(message, e);
+        return false;
+      }
+    }
+  }
+
+  static class FileSystemLs extends FileSystemFunction {
+    private static ThreadLocal<DateFormat> dateFormat = new ThreadLocal<DateFormat>() {
+
+      @Override
+      protected DateFormat initialValue() {
+        return DateFormat.getDateTimeInstance(
+                DateFormat.DEFAULT,
+                DateFormat.DEFAULT,
+                Locale.getDefault()
+                );
+      }
+    };
+
+    FileSystemLs(FileSystemGetter getter) {
+      super(getter);
+    }
+
+    @Override
+    public Object apply(List<Object> args, Context context) throws ParseException {
+
+      String path = (String) args.get(0);
```

Comment:
Might a sensible default be better here when no args are passed? Perhaps the user's home dir or '/'?
```
[Stellar]>>> HDFS_LS()
[!] Unable to execute: Index: 0, Size: 0
org.apache.metron.common.dsl.ParseException: Unable to execute: Index: 0, Size: 0
...
Caused by: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
       	at java.util.LinkedList.checkElementIndex(LinkedList.java:555)
       	at java.util.LinkedList.get(LinkedList.java:476)
       	at org.apache.metron.management.FileSystemFunctions$FileSystemLs.apply(FileSystemFunctions.java:213)
```